crates/vm/levm/src/gas_cost.rs

```diff
@@ -276,49 +300,52 @@ fn mem_expansion_behavior(
         .ok_or(OutOfGasError::GasCostOverflow)
 }
 
-pub fn sload(is_cached: bool) -> U256 {
-    if is_cached {
-        // If slot is warm (cached) add 100 to base_dynamic_gas
-        WARM_ADDRESS_ACCESS_COST
+pub fn sload(storage_slot_was_cold: bool) -> Result<U256, VMError> {
+    let static_gas = SLOAD_STATIC;
+
+    let dynamic_cost = if storage_slot_was_cold {
+        SLOAD_COLD_DYNAMIC
     } else {
-        // If slot is cold (not cached) add 2100 to base_dynamic_gas
-        COLD_STORAGE_ACCESS_COST
-    }
+        SLOAD_WARM_DYNAMIC
+    };
+
+    Ok(static_gas
+        .checked_add(dynamic_cost)
+        .ok_or(OutOfGasError::GasCostOverflow)?)
 }
 
 pub fn sstore(
-    value: U256,
-    is_cached: bool,
     storage_slot: &StorageSlot,
-) -> Result<U256, OutOfGasError> {
-    let mut base_dynamic_gas: U256 = U256::zero();
-
-    if !is_cached {
-        // If slot is cold 2100 is added to base_dynamic_gas
-        base_dynamic_gas = base_dynamic_gas
-            .checked_add(U256::from(2100))
-            .ok_or(OutOfGasError::GasCostOverflow)?;
-    };
+    new_value: U256,
+    storage_slot_was_cold: bool,
+) -> Result<U256, VMError> {
+    let static_gas = SSTORE_STATIC;
 
-    let sstore_gas_cost = if value == storage_slot.current_value {
+    let mut base_dynamic_gas = if new_value == storage_slot.current_value {
         U256::from(100)
     } else if storage_slot.current_value == storage_slot.original_value {
-        if storage_slot.original_value == U256::zero() {
+        if storage_slot.original_value.is_zero() {
             U256::from(20000)
         } else {
-            U256::from(2900)
+            U256::from(5000)
         }
     } else {
         U256::from(100)
     };
 
-    base_dynamic_gas
-        .checked_add(sstore_gas_cost)
-        .ok_or(OutOfGasError::GasCostOverflow)
+    if storage_slot_was_cold {
+        base_dynamic_gas = base_dynamic_gas
+            .checked_add(U256::from(2100))
```

Comment:
Can't we use constants from `gas_cost.rs` for these and the other gas costs we are adding/substracting?

Reply:
Yes!

Reply:
Done in [0835db6](https://github.com/lambdaclass/ethrex/pull/1314/commits/0835db6f1c6bca77cc429ad599e584a2094a4e5b) and [d78436a](https://github.com/lambdaclass/ethrex/pull/1314/commits/d78436a9feead6a263b05283a58cd34eb45f255b)